docs/data/pages.ts

```diff
@@ -237,7 +237,26 @@ const pages: MuiPage[] = [
         ],
       },
       { pathname: '/x/react-date-pickers/validation' },
-      { pathname: '/x/react-date-pickers/localization' },
+      {
+        pathname: '/x/react-date-pickers/localization',
+        title: 'Localization / date manipulation',
+        scopePathnames: [
+          '/x/react-date-pickers/adapters',
+          '/x/react-date-pickers/adapters-locale',
+          '/x/react-date-pickers/localization',
+        ],
+        children: [
+          { pathname: '/x/react-date-pickers/adapters', title: 'Library to manipulate dates' },
+          {
+            pathname: '/x/react-date-pickers/adapters-locale',
+            title: 'Localized dates',
+          },
+          {
+            pathname: '/x/react-date-pickers/localization',
+            title: 'Translated components',
+          },
```

Comment:
Related to https://github.com/mui/mui-x/pull/6625#pullrequestreview-1189602404

@alexfauquette I propose to have the discussion about where each peace if do should live here, to have a thread :+1: 

> The introduction to adapter needs to be done in the getting started because they are needed to get started with the library.

My approach was to put a concise version of "Library to manipulate date" in the "Getting Started" page and then to have a full page to discuss the trade-offs, the specificity of each library etc...
The main addition of the "Library to manipulate date" is the non-Gregorian calendars, which @alexfauquette proposs  to move to a standalone page "Calendar sytems".

If we do so, we just have to make  sure that the content of "Getting Started" is not to long.
With what we currently have in "Library to manipulate dates", it's probably fine.

Reply:
Moreover, it sounds interesting that all users have faced the content of **Why do you need a library?** which is short and important to get the philosophy of the package

Reply:
Done :+1: